### DISCUSSION TITLE
Backspace deletion has become basically unusable in v18.1

### DISCUSSION THREAD

**Discussion by i0ntempest** [will-stone/browserosaurus#530] [General]
In v18.1 where the URL bar no longer shows the full URL, using backspace to manipulate URLs has become a pain since I cannot preview URL in real time. I use this feature a lot since I come into a lot of URLs with tracking stuff at the end. I have always thought the URL bar showing the full URL is an integral part of a utility like this, and v18.1 has destroyed a very useful feature.

If I were to propose a solution, it would be letting the URL bar expand and show the full URL on mouse cursor contact (and/or on the first backspace hit), then subsequent backspace hits just deletes bits like before.

**Comment by will-stone** [comment]
Hmmm, thanks @i0ntempest. It's a tricky one as it's a feature I've _never_ used, not once, and I'm surprised anyone does. But I'm open to finding a solution. Can you please use a doodle tool like https://excalidraw.com/ to figure out the UX of a solution? As I'm not sure hovering would work, as it could be easily accidentally triggered when trying to mouse over the bottom browser.

**Comment by i0ntempest** [comment]
> it could be easily accidentally triggered when trying to mouse over the bottom browser.

Good point. In that case I think you could just use the first backspace hit to expand the URL, and use mouse hovering for the temporary popup like right now. In my use case the URLs with tracking generally come from WeChat and Baidu, because they want every bit of your data for fingerprinting.

**Comment by i0ntempest** [comment]
Or, maybe we could let the first mouse click on the URL bar (rather than just hovering) expand it?

**Comment by will-stone** [comment]
@i0ntempest I'm thinking of removing the backspace functionality completely, in favour of automatic URL cleaning using this package: https://github.com/DrKain/tidy-url What do you reckon? Or do you know of any other URL cleaners?

**Comment by i0ntempest** [comment]
@will-stone Damn I’ve no idea such thing existed. If it does a decent job at cleaning URLs then certainly. I’ll try a bit with the URLs I have and get back to you.

**Reply by will-stone** [reply]
Awesome, thanks. I’ve tried it out and it seems to work pretty well so far. Any you find don’t work, it might be worth submitting an issue/PR to that project.

**Reply by i0ntempest** [reply]
Hey, sorry for taking so long. I just got time to test this package. My test results are... mixed. It works with some and does not with some others. Probably because it uses a list of known websites and everything not in that list is just ignored. So this package is a welcomed addition, but I'd prefer if we could retain and fix the backspace deletion function.

**Reply by will-stone** [reply]
Hi @i0ntempest, thanks for doing some testing. Unfortunately, as I said before, it's a feature I've just never used, and so this current solution is the best I can offer. This helps me keep maintenance costs (time) down, and able to work on other features or fixes. It looks like the `tidy-url` package is a well maintained project though, and I'm sure they'd welcome any PRs for supporting more URLs if you'd like to help them out 😊 If they do a release that supports something you need, feel free to bump the package here or give me a shout and I can take a look at including it in the next Browserosaurus release.

**Reply by DrKain** [reply]
Hi Will, thanks for giving the package a shot and I appreciate the feedback.

Just letting you know there have been a fair few bug fixes and rule updates since `1.3.4` and it might be worth bumping the package to avoid any issues down the line, or just to get the 150+ new parameters.   

As an additional note, is there an opt-out option in browserosaurus? While I'm sure most people would appreciate the url cleaning I feel the option should still be available for people to enable/disable if they want to.

**Reply by will-stone** [reply]
Hi @DrKain 👋 Thanks for dropping by, and thanks for maintaining the tidy url package; I’ll be sure to update the package in the next release 🙂 

No opt out at the moment but it shouldn’t be too hard to add, I can see why some people may not want this feature, or more to the point may not know it’s there. I’ll look at adding that too 😁
